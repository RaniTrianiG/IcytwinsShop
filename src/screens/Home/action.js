/* eslint-disable import/prefer-default-export */
import { ACTIONS, API } from '../../constants';
import { AXIOS } from '../../configs';

export const getHomeData = () => dispatch => {
  dispatch({ type: ACTIONS.GET_HOME_START });
  AXIOS.get(API.POSTS)
    .then(resp => {
      dispatch({ type: ACTIONS.GET_HOME_SUCCESS, data: resp });
    })
    .catch(error => {
      dispatch({ type: ACTIONS.GET_HOME_FAILED, error });
    });
};
