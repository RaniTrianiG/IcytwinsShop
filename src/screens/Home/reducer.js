import { ACTIONS } from '../../constants';
import { objectUpdater } from '../../utils';

const initialState = {
  isLoading: false,
  error: null,
  data: null
};

const reducers = (state = initialState, action) => {
  const { data, error } = action;

  switch (action.type) {
    case ACTIONS.GET_HOME_START:
      return objectUpdater(state, { isLoading: true });
    case ACTIONS.GET_HOME_SUCCESS:
      return objectUpdater(state, { isLoading: false, data });
    case ACTIONS.GET_HOME_FAILED:
      return objectUpdater(state, { isLoading: false, error });
    default:
      return state;
  }
};

export default reducers;
