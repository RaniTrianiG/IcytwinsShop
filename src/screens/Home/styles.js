import { StyleSheet, Dimensions } from 'react-native';

const WIDTH = Dimensions.get('screen').width - 28 * 2;

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F9F9F9',
    flex: 1
  },
  content: {
    flex: 1,
    marginTop: 13,
    paddingHorizontal: 26
  },

  icon: {
    flex: 1,
    height: null,
    resizeMode: 'contain',
    width: null
  },

  menu: {
    alignItems: 'center',
    backgroundColor: '#fff',
    borderRadius: 10,
    elevation: 2,
    height: WIDTH / 3 - 14,
    justifyContent: 'center',
    marginBottom: 13,
    width: WIDTH / 3 - 10
  },
  menuContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between'
  },
  menuDesc: {
    color: '#888888',
    fontSize: 11,
    fontWeight: '400',
    textAlign: 'center'
  },
  menuDescTwo: {
    fontSize: 15,
    fontWeight: '700',
    lineHeight: 15,
    textAlign: 'center'
  },
  menuIcon: {
    aspectRatio: 1,
    width: 65
  },
  menuLong: {
    flexDirection: 'row',
    paddingHorizontal: 13,
    width: (WIDTH / 3) * 2 - 5
  },
  menuLongDesc: {
    fontSize: 11,
    fontWeight: '400'
  },
  menuLongNum: {
    color: '#13A310',
    fontSize: 20,
    fontWeight: '700'
  },
  menuMedium: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingRight: 12,
    width: WIDTH / 2 - 8
  },
  menuName: {
    fontSize: 11,
    fontWeight: '700',
    textAlign: 'center'
  },
  menuNum: {
    fontSize: 32,
    fontWeight: '400',
    textAlign: 'center'
  },
  title: {
    fontFamily: 'serif',
    fontSize: 18,
    fontWeight: '700',
    lineHeight: 23.85,
    marginBottom: 19,
    marginTop: 20
  }
});

export default styles;
