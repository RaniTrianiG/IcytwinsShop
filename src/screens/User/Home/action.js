/* eslint-disable no-undef */
/* eslint-disable import/prefer-default-export */
import { API } from 'react-native-dotenv';
import { ACTIONS, API as URLS } from '../../../constants';
import { reqOption } from '../../../utils';

export const getData = (categoryId = null, successCallback) => async dispatch => {
  const defaultOpt = await reqOption();
  dispatch({ type: ACTIONS.GET_PRODUCT_START });

  const options = {
    method: 'get',
    headers: defaultOpt.headers
  };

  const apiUrl =
    categoryId != null ? `${API + URLS.PRODUCT}?category_id=${categoryId}` : `${API + URLS.PRODUCT}`;
  console.log(apiUrl);
  fetch(apiUrl, options)
    .then(response =>
      response.text().then(resData => ({
        data: resData === '' ? {} : JSON.parse(resData),
        status: true,
        httpStatus: response.status
      }))
    )
    .then(res => {
      const { data } = res;

      dispatch({ type: ACTIONS.GET_PRODUCT_SUCCESS, data });
      successCallback(res);
    })
    .catch(error => {
      dispatch({ type: ACTIONS.GET_PRODUCT_FAILED, error });
    });
};
