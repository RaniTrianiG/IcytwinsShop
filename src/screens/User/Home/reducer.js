import { ACTIONS } from '../../../constants';
import { objectUpdater } from '../../../utils';

const initialState = {
  data: null
};

const reducers = (state = initialState, action) => {
  const { data } = action;

  switch (action.type) {
    case ACTIONS.GET_PRODUCT_SUCCESS:
      return objectUpdater(state, { data });
    default:
      return state;
  }
};

export default reducers;
