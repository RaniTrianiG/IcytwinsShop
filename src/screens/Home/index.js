import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from './action';
import { getProfile } from '../Login/action';
import Component from './component';

const mapStateToProps = state => {
  return {
    ...state.home
  };
};

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators({ ...actions, getProfile }, dispatch)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Component);
