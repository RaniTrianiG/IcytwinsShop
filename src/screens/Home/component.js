import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { Navigation } from 'react-native-navigation';
import { API } from 'react-native-dotenv';
import styles from './styles';
import { SCREENS } from '../../constants';
import { Button } from 'react-native';

class ExamplePage extends React.Component {
  componentDidMount() {
    const { actions } = this.props;
    actions.getHomeData();
  }

  navigateToAuth = () => {
    const { componentId} = this.props;
    Navigation.push(componentId, {
      component: {
        name: SCREENS.EXAMPLE,
        options: {
          statusBar: {
            style: 'dark',
            backgroundColor: '#f2f2f2'
          }
        }
      }
    })
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>This is home screen with api {API}</Text>
        <Button onPress={this.navigateToAuth}
        title="Go To Example"/>
      </View>
    );
  }
}

ExamplePage.defaultProps = {
  actions: {}
};

ExamplePage.propTypes = {
  actions: PropTypes.object
};

export default ExamplePage;
