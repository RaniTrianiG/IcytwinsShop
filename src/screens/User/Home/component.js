/* eslint-disable camelcase */
/* eslint-disable react/jsx-boolean-value */
import React from 'react';
import { View, Text, Image, ScrollView, TouchableOpacity, FlatList } from 'react-native';

import { Navigation } from 'react-native-navigation';
import { API } from 'react-native-dotenv';

import PropTypes from 'prop-types';
import { SCREENS } from '../../../constants';

import IconHome from '../../../assets/png/icon-home-red.png';
import IconBag from '../../../assets/png/icon-bag.png';
import Banner from '../../../assets/png/banner.png';
import IconUser from '../../../assets/png/icon-user.png';
import IconHistory from '../../../assets/png/icon-history.png';

import styles from './styles';

class Home extends React.Component {
  componentDidMount() {
    const { actions, item } = this.props;

    actions.getData(item?.id ?? null);
  }

  _navigateToDetail = item => () => {
    const { componentId } = this.props;

    Navigation.push(componentId, {
      component: {
        name: SCREENS.USER_DETAIL_PRODUCT,
        passProps: {
          selectedProduct: item
        },
        options: {
          statusBar: {
            style: 'dark',
            backgroundColor: '#F9F9F9'
          }
        }
      }
    });
  };

  _navigateToCategory = () => {
    const { componentId } = this.props;

    Navigation.push(componentId, {
      component: {
        name: SCREENS.CATEGORYUSER,
        options: {
          statusBar: {
            style: 'dark',
            backgroundColor: '#F9F9F9'
          }
        }
      }
    });
  };

  _handleTabBtnPress = route => () => {
    Navigation.setRoot({
      root: {
        stack: {
          children: [
            {
              component: {
                name: route,
                options: {
                  statusBar: {
                    style: 'dark',
                    backgroundColor: '#F9F9F9',
                    drawBehind: false
                  }
                }
              }
            }
          ],
          options: {
            topBar: {
              visible: false
            }
          }
        }
      }
    });
  };

  _renderItem = ({ item }) => {
    const thumbnail = item.imgs.find(({ is_thumbnail }) => is_thumbnail);
    const imgUrl = `${API}/storage/${item.id}/${thumbnail?.img ?? ''}`;

    return (
      <View>
        <TouchableOpacity
          onPress={this._navigateToDetail(item)}
          style={{
            overflow: 'hidden',
            width: 150,
            aspectRatio: 1,
            backgroundColor: '#aaa',
            marginRight: 16,
            borderRadius: 5
          }}
        >
          <Image
            style={{ width: null, height: null, flex: 1, resizeMode: 'cover' }}
            source={{ uri: imgUrl }}
          />
        </TouchableOpacity>
        <Text>{item.name}</Text>
      </View>
    );
  };

  render() {
    const { data, item } = this.props;

    return (
      <View style={styles.container}>
        <View style={{ flex: 1 }}>
          <ScrollView nestedScrollEnabled={true}>
            <View
              style={{
                width: '100%',
                height: 534,
                backgroundColor: '#aaaaaa',
                justifyContent: 'flex-end',
                paddingVertical: 32,
                paddingHorizontal: 12,
                alignItems: 'flex-start'
              }}
            >
              <View style={{ position: 'absolute', top: 0, bottom: 0, right: 0, left: 0 }}>
                <Image source={Banner} style={{ width: null, height: null, resizeMode: 'cover', flex: 1 }} />
              </View>

              <TouchableOpacity
                onPress={this._handleTabBtnPress(SCREENS.REGISTER)}
                style={{ position: 'absolute', top: 12, right: 12 }}
              >
                <Text style={{ fontSize: 14, fontWeight: '700', color: '#fff' }}>Log Out</Text>
              </TouchableOpacity>

              <Text style={{ fontSize: 48, lineHeight: 59, fontWeight: 'bold', color: '#ffff' }}>
                Bouquet sale
              </Text>

              <TouchableOpacity
                onPress={this._navigateToCategory}
                style={{
                  backgroundColor: '#DB3022',
                  minWidth: 160,
                  minHeight: 36,
                  alignItems: 'center',
                  justifyContent: 'center',
                  marginTop: 12,
                  borderRadius: 36
                }}
              >
                <Text style={{ color: '#fff', fontFamily: 'serif', fontSize: 14, fontWeight: '700' }}>
                  Check
                </Text>
              </TouchableOpacity>
            </View>

            <View>
              <View
                style={{ marginHorizontal: 14, marginTop: 33, flexDirection: 'row', alignItems: 'center' }}
              >
                <View style={{ flex: 1 }}>
                  <Text style={{ fontSize: 34, fontWeight: 'bold', lineHeight: 45.05 }}>
                    {item?.name ?? 'New'}
                  </Text>
                  <Text style={{ fontFamily: 'serif', fontSize: 11, color: '#9B9B9B' }}>
                    {item?.description ?? 'Youâ€™ve never seen it before!'}
                  </Text>
                </View>
              </View>

              <FlatList
                horizontal={true}
                nestedScrollEnabled={true}
                data={data?.data ?? []}
                keyExtractor={(_, idx) => `item-${idx}`}
                renderItem={this._renderItem}
                showsHorizontalScrollIndicator={false}
                contentContainerStyle={{ marginHorizontal: 14, marginVertical: 22 }}
              />
            </View>
            <View />
          </ScrollView>
        </View>
        <View
          style={{
            flexDirection: 'row',
            backgroundColor: '#fff',
            paddingHorizontal: 26,
            justifyContent: 'space-evenly',
            paddingVertical: 12,
            borderRadius: 15
          }}
        >
          <TouchableOpacity
            disabled={true}
            onPress={this._handleTabBtnPress(SCREENS.USER_HOME)}
            style={{ alignItems: 'center' }}
          >
            <View style={{ width: 30, aspectRatio: 1, marginBottom: 5 }}>
              <Image
                style={{ width: null, height: null, flex: 1, resizeMode: 'contain' }}
                source={IconHome}
              />
            </View>
            <Text style={{ fontSize: 10, lineHeight: 10, color: '#DB3022' }}>Home</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={this._handleTabBtnPress(SCREENS.USER_CART)}
            style={{ alignItems: 'center' }}
          >
            <View style={{ width: 30, aspectRatio: 1, marginBottom: 5 }}>
              <Image style={{ width: null, height: null, flex: 1, resizeMode: 'contain' }} source={IconBag} />
            </View>
            <Text style={{ fontSize: 10, lineHeight: 10 }}>Bag</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={this._handleTabBtnPress(SCREENS.HISTORY)}
            style={{ alignItems: 'center' }}
          >
            <View style={{ width: 30, aspectRatio: 1, marginBottom: 5 }}>
              <Image
                style={{ width: null, height: null, flex: 1, resizeMode: 'contain' }}
                source={IconHistory}
              />
            </View>
            <Text style={{ fontSize: 10, lineHeight: 10 }}>History</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={this._handleTabBtnPress(SCREENS.PROFILE_USER)}
            style={{ alignItems: 'center' }}
          >
            <View style={{ width: 30, aspectRatio: 1, marginBottom: 5 }}>
              <Image
                style={{ width: null, height: null, flex: 1, resizeMode: 'contain' }}
                source={IconUser}
              />
            </View>
            <Text style={{ fontSize: 10, lineHeight: 10 }}>Profile</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

Home.defaultProps = {
  componentId: 'usehomescreen',
  actions: {},
  data: {},
  item: {}
};

Home.propTypes = {
  componentId: PropTypes.string,
  actions: PropTypes.object,
  data: PropTypes.object,
  item: PropTypes.object
};

export default Home;
