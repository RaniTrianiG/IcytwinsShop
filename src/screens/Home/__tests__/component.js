import React from 'react';
import { Button } from 'react-native';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { reducer as formReducer } from 'redux-form';

import renderer from 'react-test-renderer';
import Home from '../component';

jest.mock('@react-native-community/netinfo');
jest.mock('react-native-navigation');

describe('Home Tests', () => {
  const mockStore = configureMockStore([thunk]);
  let initialState;
  let store;
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      actions: {
        getHomeData: jest.fn()
      },
      isLoading: false
    };
    initialState = {
      form: formReducer
    };
    wrapper = shallow(<Home {...props} />);
  });

  afterEach(() => {
    store.clearActions();
  });

  it('renders correctly with initial state', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test home button', () => {
    const spy = jest.spyOn(wrapper.instance(), 'navigateToAuth');
    wrapper.instance().forceUpdate();
    wrapper
      .find(Button)
      .at(0)
      .props()
      .onPress();
    expect(spy).toBeCalled();
  });
});
