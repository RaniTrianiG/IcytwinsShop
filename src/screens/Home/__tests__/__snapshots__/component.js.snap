// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests renders correctly with initial state 1`] = `
<View
  style={
    Object {
      "borderWidth": 1,
    }
  }
>
  <Text>
    This is home screen with api 
    https://jsonplaceholder.typicode.com
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go To Example
      </Text>
    </View>
  </View>
</View>
`;
